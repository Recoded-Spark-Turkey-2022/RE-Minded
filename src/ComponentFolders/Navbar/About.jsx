import { Link } from 'react-router-dom';
import { AiFillCaretDown } from "react-icons/ai";
import "./styl.css"
import { useTranslation } from 'react-i18next';




const About = () => {
    const { t } = useTranslation();
    return (
        <div>
            <div className="dropdown">
                <div className=' flex '>
                <button type='button'>{t('navbar.about')}</button><AiFillCaretDown className=' mt-1 ml-2'/>
                </div>
                <div className="dropdown-content">
                    <Link to="/About">
                        <a href="About">{t('about_dropdown.about')}</a>
                    </Link>
                    <Link to="/Team">
                        <a href="Team">{t('about_dropdown.team')}</a>
                    </Link>
                    <Link to="/careers">
                        <a href="careers">{t('about_dropdown.careers')}</a>
                    </Link>
                </div>
            </div>
        </div>
    )

}



export default About;



