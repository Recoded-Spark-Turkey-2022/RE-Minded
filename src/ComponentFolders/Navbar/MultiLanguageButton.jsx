import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import i18n from '../../i18n'; 

function MultiLangDropDown() {
  const { t } = useTranslation();

  const [locale, setLocale] = useState('en')
  const handleChangeLocale = (lan) => {
    setLocale(lan);
    i18n.changeLanguage(lan);
}


  return (
    <div>
    <select id="languages" value={locale} onChange={(e) => handleChangeLocale(e.target.value)}>
    <option value="en">{t('languages.en')}</option>
    <option value="tr">{t('languages.tr')}</option>
    </select>
    </div>
  );
}

export default MultiLangDropDown;