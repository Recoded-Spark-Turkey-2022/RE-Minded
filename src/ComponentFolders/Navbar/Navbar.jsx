import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Logo from './Images/Logo.svg';

export default function NavBar({handleSignout}) {

  const [navbar, setNavbar] = useState(false);

  const signoutButton = useSelector((state) => state.signoutButton)

  return (
    <nav className="w-full sticky top-0 z-50 bg-cyan-50 shadow font-poppins">
      <div className="justify-between px-4 mx-auto  md:items-center md:flex md:px-8">
        <div>
          <div className="flex items-center  py-3 md:py-5 md:block">
            <Link to="/">
              <div className="flex flex-row ">
                <img src={Logo} alt="Logo" />
                <a href="Logo">
                  <h2 className="text-3xl text-bold m-4 ml-3 font-medium">
                    Healing
                  </h2>
                </a>
              </div>
            </Link>
            <div className="md:hidden">
              <button
                type="button"
                className="p-2 text-gray-700 rounded-md outline-none focus:border-gray-400 focus:border"
                onClick={() => setNavbar(!navbar)}
              >
                {navbar ? (
                  <svg
                    className="w-6 h-6 "
                    viewBox="0 0 20 20"
                    fill="currentColor"
                  >
                    <path
                      fillRule="evenodd"
                      d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                      clipRule="evenodd"
                    />
                  </svg>
                ) : (
                  <svg
                    className="w-6 h-6 "
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                    strokeWidth={2}
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      d="M4 6h16M4 12h16M4 18h16"
                    />
                  </svg>
                )}
              </button>
            </div>
          </div>
        </div>
        <div>
          <div
            className={`flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0 ${
              navbar ? 'block' : 'hidden'
            }`}
          >
            <div className="mt-3 space-y-2 lg:hidden md:hidden ">
              <div className="flex flex-col  items-center  text-xl ">
                <Link
                  to="/"
                  className=" hover:text-indigo-200 pb-1 "
                  href="Home"
                >
                  Home
                </Link>
                <Link
                  to="blog"
                  className=" hover:text-indigo-200 p-2"
                  href="Blog"
                >
                  Blog
                </Link>
                <Link
                  to="about"
                  className=" hover:text-indigo-200 p-2"
                  href="Home"
                >
                  About
                </Link>
                <Link
                  to="contactus"
                  className=" hover:text-indigo-200 p-2"
                  href="Blog"
                >
                  Contact Us
                </Link>
                <Link to={signoutButton ? 'signup' : 'login'}>
                  {!signoutButton ? (
                    <a
                      href="Log In"
                      className="flex justify-center px-4 py-2 text-center rounded-md shadowtransition-all duration-250 bg-cyan-400 hover:bg-cyan-500 text-xl"
                    >
                      Log In
                    </a>
                  ) : (
                    <button
                      type="button"
                      className="flex justify-center px-4 py-2 text-center rounded-md shadowtransition-all duration-250 bg-cyan-400 hover:bg-cyan-500 text-xl"
                      onClick={handleSignout}
                    >
                      Sign out
                    </button>
                  )}
                </Link>
              </div>
            </div>
          </div>
        </div>
        <div className="hidden space-x-2 md:inline-block">
          <ul className="items-center justify-center space-y-8 md:flex md:space-x-6 md:space-y-0">
            <Link to="/">
              <li className=" hover:text-indigo-200">
                <a href="Home">Home</a>
              </li>
            </Link>
            <Link to="blog">
              <li className=" hover:text-indigo-200">
                <a href="Blog">Blog</a>
              </li>
            </Link>

            <li className=" hover:text-indigo-200 cursor-pointer">
              <select className=" p-2.5 bg-cyan-50 cursor-pointer">
                <option>
                  <Link to="about">
                    <li>
                      <a href="About">About</a>
                    </li>
                  </Link>
                </option>
                <option>
                  <Link to="team" href="Team">
                    Team
                  </Link>
                </option>
                <option>
                  <Link to="careers" href="Carees">
                    Carees
                  </Link>
                </option>
              </select>
            </li>

            <Link to="contactus">
              <li className=" hover:text-indigo-200">
                <a href="Contact">Contact US</a>
              </li>
            </Link>
            <Link to={signoutButton ? 'signup' : 'login'}>
              {!signoutButton ? (
                <a
                  href="Log In"
                  className="px-4 py-2 rounded-md shadow transition-all duration-250 bg-cyan-400 hover:bg-cyan-500"
                >
                  Log in
                </a>
              ) : (
                <button
                  type="button"
                  className="flex justify-center px-4 py-2 text-center rounded-md shadowtransition-all duration-250 bg-cyan-400 hover:bg-cyan-500 text-xl"
                  onClick={handleSignout}
                >
                  Sign out
                </button>
              )}
            </Link>
          </ul>
        </div>
      </div>
    </nav>
  );
}
