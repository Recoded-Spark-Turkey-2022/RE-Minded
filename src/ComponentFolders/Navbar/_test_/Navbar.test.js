import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

test('Navbar test snapshot', () => {
  const component = renderer.create(<Navbar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
