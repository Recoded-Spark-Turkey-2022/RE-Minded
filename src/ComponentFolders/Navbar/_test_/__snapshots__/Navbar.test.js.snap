// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test snapshot 1`] = `
<nav

  className="w-full bg-cyan-50 shadow font-poppins"
>
  <div
    className="justify-between px-4 mx-auto  md:items-center md:flex md:px-8"
  >
    <div>
      <div
        className="flex items-center  py-3 md:py-5 md:block"
      >
        <div
          className="flex flex-row "
        >
          <img
            alt="Logo"
            src="Logo.svg"
          />
          <a
            href="Logo"
          >
            <h2
              className="text-3xl text-bold m-4 ml-3 font-medium"
            >
              Healing
            </h2>
          </a>
        </div>
        <div
          className="md:hidden"
        >
          <button
            className="p-2 text-gray-700 rounded-md outline-none focus:border-gray-400 focus:border"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="w-6 h-6 "
              fill="none"
              stroke="currentColor"
              strokeWidth={2}
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        className="flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0 hidden"
      >
        <div
          className="mt-3 space-y-2 lg:hidden md:hidden "
        >
          <div
            className="flex flex-col  items-center  text-xl "
          >
            <a
              className=" hover:text-indigo-200 pb-1 "
              href="Home"
            >
              Home
            </a>
            <a
              className=" hover:text-indigo-200 p-2"
              href="Blog"
            >
              Blog
            </a>
            <a
              className=" hover:text-indigo-200 p-2"
              href="Home"
            >
              About
            </a>
            <a
              className=" hover:text-indigo-200 p-2"
              href="Blog"
            >
              Contact Us
            </a>
            <a
              className="flex justify-center w-1/3  px-4 py-2 text-center rounded-md shadowtransition-all duration-250 bg-cyan-400 hover:bg-cyan-500 text-xl"
              href="Log In"
            >
              Log In
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="hidden space-x-2 md:inline-block"
    >
      <ul
        className="items-center justify-center space-y-8 md:flex md:space-x-6 md:space-y-0"
      >
        <li
          className=" hover:text-indigo-200"
        >
          <a
            href="Home"

          >
            Home
          </a>
        </li>
        <li

          className=" hover:text-indigo-200"
        >
          <a
            href="Blog"
          >
            Blog
          </a>
        </li>
        <li
          className=" hover:text-indigo-200 cursor-pointer"
        >
          <select
            className=" p-2.5 bg-cyan-50 cursor-pointer"
          >
            <option>
              <a
                href="About"

              >
                About
              </a>
            </option>
            <option>

              <a
                href="Team"
              >
                Team
              </a>
            </option>
            <option>
              <a
                href="Carees"
              >
                Carees

              </a>
            </option>
          </select>
        </li>
        <li

          className=" hover:text-indigo-200"
        >
          <a
            href="Contact"
          >
            Contact US
          </a>
        </li>
        <a
          className="px-4 py-2 rounded-md shadow transition-all duration-250 bg-cyan-400 hover:bg-cyan-500"
          href="Log In"
        >
          Log in
        </a>

      </ul>
    </div>
  </div>
</nav>
`;
